spikeinterface/toolkit/preprocessing/filter.py

```diff
@@ -63,33 +64,42 @@ def __init__(self, recording, band=[300., 6000.], btype='bandpass',
         # self.coeff is 'sos' or 'ab' style
         coeff = scipy.signal.iirfilter(N, Wn, analog=False, btype=btype, ftype=ftype, output=filter_mode)
 
-        BasePreprocessor.__init__(self, recording, dtype=dtype)
-        dtype_base = self.get_dtype()
+        if dtype is None:
+            dtype_base = recording.get_dtype()
+        else:
+            dtype_base = dtype
+        dtype_base, force_traces_dtype = _correct_uint(dtype_base)
+
+        BasePreprocessor.__init__(self, recording, dtype=dtype_base)
         self.annotate(is_filtered=True)
 
         margin = int(margin_ms * sf / 1000.)
         for parent_segment in recording._recording_segments:
             self.add_recording_segment(FilterRecordingSegment(parent_segment, coeff, filter_mode, margin,
-                                                              dtype_base))
+                                                              dtype_base, force_traces_dtype))
 
         self._kwargs = dict(recording=recording.to_dict(), band=band, btype=btype,
                             filter_order=filter_order, ftype=ftype, filter_mode=filter_mode, margin_ms=margin_ms)
 
 
 class FilterRecordingSegment(BasePreprocessorSegment):
-    def __init__(self, parent_recording_segment, coeff, filter_mode, margin, dtype):
+    def __init__(self, parent_recording_segment, coeff, filter_mode, margin, dtype, force_traces_dtype):
```

Comment:
I think here dtype and force dtype is eactly the same concept.
I would leave dtype only.